File: src/demo_greeting/greeting.py
# Implements: REQ-200, REQ-201, REQ-202 (docs/specs/2026-05-28-demo-greeting-design.md)
def greet(name: str) -> str:
    if name == "":
        raise ValueError("name must be non-empty")
    return f"Hello, {name}!"

File: src/DemoGreeting/Greeting.cs
// Implements: REQ-200, REQ-201, REQ-202 (docs/specs/2026-05-28-demo-greeting-design.md)
namespace DemoGreeting;

public static class Greeting
{
    public static string Greet(string name)
    {
        if (name == "")
        {
            throw new ArgumentException("name must be non-empty");
        }
        return $"Hello, {name}!";
    }
}
